src/contracts/DatumRegistry.ts

```diff
@@ -0,0 +1,67 @@
+import { Contract, ethers } from 'ethers';
+
+import { IDatumCreation } from '../types/IDatumCreation';
+import { datumRegistryAbi } from './abi';
+import { LeafKeyCoder } from '../index';
+
+export class DatumRegistryContract {
+  contract: Contract;
+
+  constructor(providerOrSigner: ethers.providers.Provider | ethers.Signer, datumRegistryAddress: string) {
+    this.contract = new ethers.Contract(datumRegistryAddress, datumRegistryAbi, providerOrSigner);
+  }
+
+  async create(createParams: IDatumCreation): Promise<ethers.providers.TransactionResponse> {
+    const funderAddress = createParams.funderAddress ?? (await this.contract.signer.getAddress());
+    const encodedKeys = this.encodeKeys(createParams.keys);
+
+    const tx = await this.contract.create(
+      createParams.receiverAddress,
+      funderAddress,
+      encodedKeys,
+      createParams.depositAmount
+    );
+
+    return tx;
+  }
+
+  async setDatumEnabled(receiverAddress: string, enabledFlag: boolean): Promise<ethers.providers.TransactionResponse> {
+    const tx = await this.contract.setDatumEnabled(receiverAddress, enabledFlag);
+    return tx;
+  }
+
+  async deposit(receiverAddress: string, depositAmount: string): Promise<ethers.providers.TransactionResponse> {
+    const tx = await this.contract.deposit(receiverAddress, depositAmount);
+    return tx;
+  }
+
+  async withdraw(receiverAddress: string, withdrawAmount?: string): Promise<ethers.providers.TransactionResponse> {
+    const amount = await (withdrawAmount
+      ? withdrawAmount
+      : this.getBalance(receiverAddress, await this.contract.signer.getAddress()));
```

Comment:
```suggestion
    const amount = withdrawAmount ?? await (this.getBalance(receiverAddress, await this.contract.signer.getAddress())));
```
Just like in the `create` method